src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -427,7 +429,11 @@ public static void threadDumpStartup() throws Exception {
         Thread t = new Thread("Support core plugin startup diagnostics") {
             @Override
             public void run() {
-                while (!milestonesCompleted) {
+                while (true) {
+                    final Jenkins jenkins =Jenkins.getInstance();
+                    if (jenkins == null || jenkins.getInitLevel() != InitMilestone.COMPLETED) {
+                        continue;
```

Comment:
Just fixed the code without performance polishing.
@christ66 and @aheritier may enhance it later